cypress-tests/cypress/e2e/v2/projectResources.cy.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  getRandomString,
+  getUserData,
+  validateLoginV2,
+} from "../../support/commands/general";
+import { ProjectIdentifierV2 } from "../../support/types/project.types";
+import { User } from "../../support/types/user.types";
+import {
+  createProjectIfMissingAPIV2,
+  deleteProjectFromAPIV2,
+  getProjectByNamespaceAPIV2,
+} from "../../support/utils/projectsV2.utils";
+
+const sessionId = ["projectResources", getRandomString()];
+
+describe("Project resources - work with code, data, environments", () => {
+  // Define some project details
+  const projectNameRandomPart = getRandomString();
+  const projectName = `project-resources-${projectNameRandomPart}`;
+  const projectDescription =
+    "This is a test project from Cypress to test working with code, data, environments";
+  const projectIdentifier: ProjectIdentifierV2 = {
+    slug: projectName,
+    id: null,
+    namespace: null,
+  };
+
+  // Create a project and keep that around for the rest of the tests
+  before(() => {
+    // Login
+    cy.session(
+      sessionId,
+      () => {
+        cy.robustLogin();
+      },
+      validateLoginV2,
+    );
+
+    // Create the project and save its deetails
+    getUserData().then((user: User) => {
+      projectIdentifier.namespace = user.username;
+      createProjectIfMissingAPIV2({
+        description: projectDescription,
+        name: projectName,
+        namespace: user.username,
+        slug: projectName,
+        visibility: "private",
+      }).then((project) => (projectIdentifier.id = project.id));
+    });
+  });
+
+  // Restore the session (login)
+  beforeEach(() => {
+    cy.session(
+      sessionId,
+      () => {
+        cy.robustLogin();
+      },
+      validateLoginV2,
+    );
+  });
+
+  // Cleanup the project after the test -- useful on failure
+  after(() => {
+    getProjectByNamespaceAPIV2(projectIdentifier).then((response) => {
+      if (response.status === 200) {
+        projectIdentifier.id = response.body.id;
+        projectIdentifier.namespace = response.body.namespace;
+        deleteProjectFromAPIV2(projectIdentifier);
+      }
+    });
+  });
+
+  const visitCurrentProject = () => {
+    // There should a link on the dashboard for a newly created project
+    cy.visit("/v2");
+    cy.getDataCy("dashboard-project-list")
+      .get(
+        `a[href*="/${projectIdentifier.namespace}/${projectIdentifier.slug}"]`,
+      )
+      .click();
+    cy.getDataCy("project-name").should("contain", projectName);
+  };
+
+  it("Work with data", () => {
```

Comment:
The title is too general that it makes me think that we tested everything - i.e. even seeing that the data connector shows up in the session. But we have not.

```suggestion
  it("Add and modify data connectors", () => {
```